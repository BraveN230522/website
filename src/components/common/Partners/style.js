import styled from '@emotion/styled';

const PartnersStyles = styled.section`
  ${({ theme }) => theme.setContainer()};
`;

const PartnerTitle = styled.header`
  opacity: 0.5;
  font-size: 22px;
  font-weight: 400;
  letter-spacing: -0.15px;
  text-align: center;
  margin: 61px 0 46px;
  color: ${({ theme }) => theme.colors.black};

  ${({ theme }) => theme.media.phone`
    margin: 40px 0 17px;
  `}
`;

const PartnersContent = styled.article`
  display: flex;
  justify-content: space-between;
  margin-bottom: 69px;
  align-items: center;
  overflow: auto;

  img {
    margin-left: 50px;
    min-width: max-content;

    &:first-of-type {
      margin-left: 0;
    }
  }

  ${({ theme }) => theme.media.phone`
    margin-bottom: 53px;
  `}
`;

export { PartnersStyles, PartnerTitle, PartnersContent };
