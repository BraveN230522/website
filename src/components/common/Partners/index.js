import React from 'react';

import BeINLogo from '@images/beIN.png';
import SingaporeRugbySevensLogo from '@images/Singapore-Rugby-7s.jpg';
import ExpressTravelLogo from '@images/express-travel-group-logo.png';
import NRLLogo from '@images/nrl-badge.png';
import UrbanAdventuresLogo from '@images/Urban-Adventures.png';
import NETSLogo from '@images/nets.jpg';

import { PartnersStyles, PartnerTitle, PartnersContent } from './style';

const Partners = () => {
  return (
    <PartnersStyles>
      <header>
        <PartnerTitle>Partners</PartnerTitle>
      </header>

      <PartnersContent>
        <img src={BeINLogo} />
        <img src={SingaporeRugbySevensLogo} />
        <img src={ExpressTravelLogo} />
        <img src={NRLLogo} />
        <img src={UrbanAdventuresLogo} />
        <img src={NETSLogo} />
      </PartnersContent>
    </PartnersStyles>
  );
};

export default Partners;
