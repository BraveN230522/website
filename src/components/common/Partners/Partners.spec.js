import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import MCGImage from '@images/mcg-partnership.jpg';
import Partners from './index';

const mockTickets = Array.from({ length: 5 }, () => ({
  average: 8.5,
  price: '129',
  discount: 20,
  title: 'Hilton Garden Inn',
  subtitle: 'Near Yankee Stadium',
  image: MCGImage,
}));

describe('<Partners />', () => {
  it('Renders the Partners unchanged', () => {
    const { container } = render(<Partners tickets={mockTickets} />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });

  it('Renders the Partners without a ticket section', () => {
    const { container } = render(<Partners />, {
      wrapper: withGlobals,
    });

    expect(container).toMatchSnapshot();
  });
});
