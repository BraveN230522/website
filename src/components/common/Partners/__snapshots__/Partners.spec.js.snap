// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Partners /> Renders the Partners unchanged 1`] = `
.emotion-4 {
  max-width: 1255px;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 30px;
  padding-right: 30px;
}

.emotion-0 {
  opacity: 0.5;
  font-size: 22px;
  font-weight: 400;
  -webkit-letter-spacing: -0.15px;
  -moz-letter-spacing: -0.15px;
  -ms-letter-spacing: -0.15px;
  letter-spacing: -0.15px;
  text-align: center;
  margin: 61px 0 46px;
  color: #000000;
}

@media (max-width:48em) {
  .emotion-0 {
    margin: 40px 0 17px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 69px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: auto;
}

.emotion-2 img {
  margin-left: 50px;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
}

.emotion-2 img:first-of-type {
  margin-left: 0;
}

@media (max-width:48em) {
  .emotion-2 {
    margin-bottom: 53px;
  }
}

<div>
  <section
    class="emotion-4 emotion-5"
  >
    <header>
      <header
        class="emotion-0 emotion-1"
      >
        Partners
      </header>
    </header>
    <article
      class="emotion-2 emotion-3"
    >
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
    </article>
  </section>
</div>
`;

exports[`<Partners /> Renders the Partners without a ticket section 1`] = `
.emotion-4 {
  max-width: 1255px;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 30px;
  padding-right: 30px;
}

.emotion-0 {
  opacity: 0.5;
  font-size: 22px;
  font-weight: 400;
  -webkit-letter-spacing: -0.15px;
  -moz-letter-spacing: -0.15px;
  -ms-letter-spacing: -0.15px;
  letter-spacing: -0.15px;
  text-align: center;
  margin: 61px 0 46px;
  color: #000000;
}

@media (max-width:48em) {
  .emotion-0 {
    margin: 40px 0 17px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 69px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: auto;
}

.emotion-2 img {
  margin-left: 50px;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
}

.emotion-2 img:first-of-type {
  margin-left: 0;
}

@media (max-width:48em) {
  .emotion-2 {
    margin-bottom: 53px;
  }
}

<div>
  <section
    class="emotion-4 emotion-5"
  >
    <header>
      <header
        class="emotion-0 emotion-1"
      >
        Partners
      </header>
    </header>
    <article
      class="emotion-2 emotion-3"
    >
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
      <img
        src="mocked-file"
      />
    </article>
  </section>
</div>
`;
